File: app/src/main/java/com/hana053/micropost/domain/RelatedUser.java
package com.hana053.micropost.domain;

import org.parceler.Parcel;
import org.parceler.ParcelConstructor;

import java.io.Serializable;

import lombok.EqualsAndHashCode;
import lombok.Value;

@EqualsAndHashCode(callSuper = false)
@Parcel
@Value
public final class RelatedUser extends User implements Serializable {

    static final long serialVersionUID = 1L;

    public final long relationshipId;

    @ParcelConstructor
    public RelatedUser(long id, String name, String email, String avatarHash, long relationshipId, UserStats userStats) {
        super(id, name, email, avatarHash, userStats);
        this.relationshipId = relationshipId;
    }

    public RelatedUser(long id, String name, String email, String avatarHash, long relationshipId) {
        this(id, name, email, avatarHash, relationshipId, null);
    }

}

File: app/src/main/java/com/hana053/micropost/domain/RelatedUser.kt
package com.hana053.micropost.domain


data class RelatedUser(
    val id: Long,
    val name: String,
    val email: String?,
    override val avatarHash: String,
    val isFollowedByMe: Boolean,
    val userStats: UserStats,
    val relationshipId: Long
) : Avatar {
    fun toUser(): User {
        return User(id, name, email, avatarHash, isFollowedByMe, userStats)
    }
}